code/game/objects/items/deployable/deployable.dm

```diff
@@ -0,0 +1,97 @@
+	var/consumed = TRUE
+	///For when consumed is false, is the carrier object currently loaded and ready to deploy its payload item?
+	var/loaded
+	///The amount of time it takes to deploy
+	var/time_to_deploy
+	///Set to true to allow deployment on top of dense objects
+	var/dense_deployment = FALSE
+
+/obj/item/deployable/attack_self(mob/user)
+	try_deploy(user, user.loc)
+
+/obj/item/deployable/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(proximity)
+		try_deploy(user, target)
+
+///Checks to see if object can deploy, either in a passed location or within its own location if none was passed and deploys if it can be.
+/obj/item/deployable/proc/try_deploy(mob/user, atom/location)
+	if(!consumed && !loaded)
+		to_chat(user, "<span class='warning'>[src] has nothing to deploy!</span>")
+		return FALSE
+	if(!location) //if no location was passed we use the current location.
+		location = loc
+	if(isopenturf(location))
+		if(dense_deployment)
+			return deploy_after(user, location)
+		else
+			var/dense_location
+			for(var/atom/movable/AM in location)
+				if(AM.density)
+					dense_location = TRUE
+					break
+			if(!dense_location)
+				return deploy_after(user, location)
+	if(user)
+		to_chat(user, "<span class='warning'>[src] can only be deployed in an open area!</span>")
+	visible_message("<span class='warning'>[src] fails to deploy!</span>")
+	return FALSE
+
+///Delays deployment for things which take time to set up
+/obj/item/deployable/proc/deploy_after(mob/user, atom/location)
+	if(!time_to_deploy || !user)
+		deploy(user, location)
+		return TRUE
+
+	user.visible_message("<span class='notice'>[user] begins to deploy [src]...</span>")
+	if(do_after(user, time_to_deploy, src))
+		deploy(user, location)
+		return TRUE
+	else
+		return FALSE
+
+///Do not call this directly, use try_deploy instead or else deployed items may end up in invalid locations
+/obj/item/deployable/proc/deploy(mob/user, atom/location)
+	if(isnull(deployed_object)) //then this must have saved contents to dump directly instead
+		for(var/atom/movable/A in contents)
+			A.forceMove(location)
+			A.add_fingerprint(user)
+	else
+		var/atom/R = new deployed_object(location)
+		for(var/atom/movable/A in contents)
+			A.forceMove(R)
+			A.add_fingerprint(user)
+		R.add_fingerprint(user)
+		if(istype(R, /obj/structure/closet/))
+			var/obj/structure/closet/sesame = R
+			sesame.open()
+	if(consumed)
+		qdel(src)
+	else
+		loaded = FALSE
+		update_icon()
+
+/obj/item/deployable/container_resist(mob/living/user)
+	if(user.incapacitated())
+		to_chat(user, "<span class='warning'>You can't get out while you're restrained like this!</span>")
+		return
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	to_chat(user, "<span class='notice'>You try to force your way out of [src]...</span>")
+	to_chat(loc, "<span class='warning'>Someone starts trying to break free of [src]!</span>")
```

Comment:
to_chat loc? What is the intention behind this (Does this work)?
If you want nearby people to see it, use visible_message instead

Reply:
That is a good question tbh.

This is taken from existing code as-is iirc so I didn't test it.

Reply:
To answer the question, yes it does work and it is actually pretty frequently used too lol

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/9547572/d3732d53-81fc-47b8-8f1d-4b922a49e8f3)